boot/bootutil/src/loader.c

```diff
@@ -1654,7 +1655,11 @@ boot_prepare_image_for_update(struct boot_loader_state *state,
          * if there is one.
          */
         BOOT_SWAP_TYPE(state) = BOOT_SWAP_TYPE_NONE;
-        return;
+        if (BOOT_EFLASH == rc)
```

Comment:
Prefer `if (rc == BOOT_EFLASH)`. Gcc will warn (which will stop the build at least in the sim) if an assignment were used instead of equality. Otherwise, this is more readable with the variable on the left.